### ISSUE TITLE
Issue with Facebook login on some devices

### ISSUE THREAD

**Issue by drdaz** [parse-community/Parse-SDK-iOS-OSX#1420]
I've got an app in the store that seems to have an issue where users can't log in using Facebook login using the FB app. Login using FB username and password works.

I've had this reported by 2 users with older slightly devices so far (iPhone 6 and iPhone SE). But I can see about 30%+ of Parse installations never get a login. Some of that could be cold feet, but could also be this issue.

Once the user has granted access through the app, the app just returns them to the screen where they can choose how to authenticate. They can then repeat the process for a sad loop.

The 'failed to login' callback isn't triggered here (I present an alert on the login controller in those cases, though I'm not sure I've ever seen that alert in development...)

There's a video of this happening here: https://www.icloud.com/iclouddrive/06N4cqtJimzR9uUJBUWnnnnAQ#FB_Login_Loop

Any ideas?

Latest Parse iOS SDK.

**Comment by brianyyz** [comment]
The latest app that I put out has about 50% email logins and 50% FB logins and they seem to be working - I've got a few Installations with no matching PFUser but I assume that is cold feet as you said. 

My dev environment is still up to date - I'll give this a whirl this evening and let you know what I find.

**Comment by drdaz** [comment]
Have you got one of those devices to test?

**Comment by brianyyz** [comment]
I've got an old 6+ that I can charge up. Hopefully it is still functional - it has been awhile!

**Comment by drdaz** [comment]
🤞🏼

**Comment by drdaz** [comment]
@brianyyz Anything? :)

**Comment by brianyyz** [comment]
No joy I'm afraid. 

I set up an iPhone 6Plus with IOS 12.3.1 with an app built with xCode 10.2.1 and the following pods
    pod 'Parse', inhibit_warnings: true
    pod 'Parse/UI'
    pod 'Parse/FacebookUtils', inhibit_warnings: true
    pod 'ReachabilitySwift', '~> 4'

I used the Log in with Facebook option in the app which followed the path to open the FB app (not the web path) on the device. Logged in with a FB test user and the login completed and the user was created with the name and email pulled from FB.

Bottom line - I can't recreate the problem. Any other ideas?

**Comment by drdaz** [comment]
Oh damn... I didn't expect that! 🥴

The only other thing I can think of is the screen size. Both of the devices I know about have smaller screens than our test devices. But I can't reproduce the issue on the Simulator with a smaller device either... so it seems really unlikely.

What version of Parse and FB libraries did CocoaPods install?

**Comment by drdaz** [comment]
The released version I'm seeing this on has version 4.44.1 for both FBSDKCoreKit and FBSDKLoginKit...

**Comment by brianyyz** [comment]
From Podfile.lock

PODS:
  - Bolts (1.9.0):
    - Bolts/AppLinks (= 1.9.0)
    - Bolts/Tasks (= 1.9.0)
  - Bolts/AppLinks (1.9.0):
    - Bolts/Tasks
  - Bolts/Tasks (1.9.0)
  - FBSDKCoreKit (4.43.0):
    - Bolts (~> 1.9)
  - FBSDKLoginKit (4.43.0):
    - FBSDKCoreKit
  - Parse (1.17.2):
    - Parse/Core (= 1.17.2)
  - Parse/Core (1.17.2):
    - Bolts/Tasks (~> 1.9)
  - Parse/FacebookUtils (1.17.2):
    - Bolts (~> 1.9)
    - FBSDKLoginKit (~> 4.33)
    - Parse/Core
  - Parse/UI (1.17.2):
    - Parse/Core
  - ReachabilitySwift (4.3.1)

**Comment by drdaz** [comment]
Interesting... minor version differences there.

Can you try adding those versions of those libs to the Podfile and reinstalling the pods? So something like:
```
pod 'FBSDKLoginKit', '~> 4.44.1'
pod 'FBSDKCoreKit', '~> 4.44.1'
```

**Comment by brianyyz** [comment]
Sure - won't get to it until later - real work is interfering :-)

**Comment by drdaz** [comment]
You're a star. Thanks :)

Would be really great if it was just an issue with that specific version 🤞🏼

**Comment by brianyyz** [comment]
Haven't forgotten about this - my time got stolen by production issues this weekend.

**Comment by drdaz** [comment]
In the mean time I'm working on getting updating Parse's FB SDK dependency updated. With some luck the issue is magically fixed in newer versions.

**Comment by drdaz** [comment]
I just pushed a test version of my app using the 5.x version of the Facebook SDK to a friend experiencing the issue. It fixes it.

So #1424 will fix this.
